dask/dataframe/groupby.py

```diff
@@ -934,8 +935,10 @@ def count(self, split_every=None, split_out=1):
 
     @derived_from(pd.core.groupby.GroupBy)
     def mean(self, split_every=None, split_out=1):
-        return (self.sum(split_every=split_every, split_out=split_out) /
-                self.count(split_every=split_every, split_out=split_out))
+        s = self.sum(split_every=split_every, split_out=split_out)
+        c = self.count(split_every=split_every, split_out=split_out)
+        c = c[s.columns]
```

Comment:
Only need this when s is a DataFrame.

LGTM other than that.

Reply:
Thanks for taking the time to review @TomAugspurger ! 

Fixed and pushed.